### PR TITLE
 Re-Write Non-conformity Notification in the Bew React UI #896 

### PR DESCRIPTION
**Pull request by EdisonMeks**
# Pull Requests Requirements

- [x] The PR title includes a brief description of the work done, including the Issue number if applicable.
- [ ] The PR includes a video showing the changes for the work done.
- [x] The PR title follows conventional commit label standards.
- [x] The changes confirm to the OpenElis Global x3 Styleguide and design documentation.
- [ ] The changes include tests or are validated by existing tests.
- [x] I have read and agree to the Contributing Guidelines of this project.

## Summary
I created a new component for the the nonconformity notification in the new react ui
## Screenshots

NonConformitypic

## Related Issue

[[Add a link to the related issue or mention it here if applicable]](https://github.com/I-TECH-UW/OpenELIS-Global-2/issues/896)

## Other

[Add any additional information or notes here]

### CODE REVIEW

**Comment by EdisonMeks** [comment]
please @mozzy11 you can review cc @sherrif10 @Agaba-derrick @reagan-meant

**Comment by mozzy11** [comment]
Thanks @EdisonMeks , 
Can you attch a screenshot or Screenrecording of your changes to help reviwers??

**Comment by EdisonMeks** [comment]
ok, let me do the needful

File: frontend/src/App.js [hunk 1]
```diff
@@ -236,6 +236,14 @@ export default function App() {
                   role=""
                   labUnitRole={{ Pathology: ["Results"] }}
                 />
+                
+                 <SecureRoute
```
[the remark this answers is not in the document]
Comment (EdisonMeks) [reply frontend/src/App.js:240]: ok

**Comment by EdisonMeks** [comment]
@mozzy11 have added the required changes you can give a review

File: frontend/src/components/reports/study/NonconformityNotification.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  const [accessionNumber, setAccessionNumber] = useState('');
+  const [selectedSite, setSelectedSite] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [notificationMessage, setNotificationMessage] = useState('');
+  const [error, setError] = useState('');
+  
+  
+  
+  const siteOptions = [
+    { label: '', value: '' },
```
Comment (mozzy11) [inline frontend/src/components/reports/study/NonconformityNotification.js:17]: you need need to pull this from this endpoint `/rest/site-names`

File: frontend/src/components/reports/study/NonconformityNotification.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    setSelectedSite(event.target.value);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (!accessionNumber) {
+      setError('Accession number is required.');
+      return;
+    }
+    setLoading(true);
```
Comment (mozzy11) [inline frontend/src/components/reports/study/NonconformityNotification.js:36]: for submititng , just dynalically build a url with params like below and openit in a new tab ie 
/ReportPrint?report=retroCInonConformityNotification&type=patient&accessionDirect=<accesionNumber>&selectList.selection=<SiteId>

Comment (mozzy11) [reply frontend/src/components/reports/study/NonconformityNotification.js:36]: see [example ](https://github.com/I-TECH-UW/OpenELIS-Global-2/blob/develop_3x/frontend/src/components/reports/routine/ReferredOut.js#L90-L105)of how its done

**Review by mozzy11 — CHANGES_REQUESTED** [review]
see comments above . 
Also attach a screen recording

**Comment by mozzy11** [comment]
Thanks @EdisonMeks .
Any progress here or blocker ??

**Comment by mozzy11** [comment]
Thanks @EdisonMeks for attemting to work on this isse. 
We needed it urgently so it has been worked on in this PR https://github.com/I-TECH-UW/OpenELIS-Global-2/pull/982.
Feel free to try out another issue

**Pull request closed without merging**